### DISCUSSION TITLE
Nextjs dealing with image galleries?

### DISCUSSION THREAD

**Discussion by denull0** [supabase/supabase#3633] [Questions]
Hey crew,

Do you have any advice on building a user gallery / photo blog? I am using Nextjs, and thinking of using Vercel for hosting. However I am a bit confused on how to deal with the image hosting situation. If I generate a public url for each photo, it's going to harm SEO for the user, right?

Thanks

**Comment by dshukertjr** [comment]
Hi @dimoFeeld!

You should be able to just use Supabase storage normally and not have any problems. 
Could you elaborate more on why generating public a url for each photo could harm SEO for the user?
